src/internal/core-api.ts

```diff
@@ -45,3 +51,112 @@ export async function getAssetTransfers(
     srcMethod
   );
 }
+
+/**
+ * This method is based on the ethers implementation of getLogs, but is expanded
+ * to support specifying an address array in the filter.
+ *
+ * The main modifications made to support an address array are:
+ * - Custom `getFilter()` method that supports an address array
+ * - Use of `arrayOf()` formatter to format the logs to avoid the `Formatter` import.
+ * - Use of `provider.send()` to avoid formatting logic in `provider.perform()`.
+ */
+export async function getLogs(
+  config: AlchemyConfig,
+  filter: Filter | FilterByBlockHash | Promise<Filter | FilterByBlockHash>
+): Promise<Array<Log>> {
+  const provider = await config.getProvider();
+  await provider.getNetwork();
+  const params = await resolveProperties({
+    filter: getFilter(config, filter)
+  });
+
+  const logs: Array<Log> = await provider.send('eth_getLogs', [params.filter]);
+  logs.forEach(log => {
+    if (log.removed == null) {
+      log.removed = false;
+    }
```

Comment:
FYI you can write this as
```ts
log.removed ??= false;
```
(If this is copied from ethers, feel free to ignore)

Reply:
ah copied from ethers, so i'll leave it as is for better searchability